Src/AutoFixture/Kernel/TypeExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Ploeh.AutoFixture.Kernel
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Reflection.Emit;
+
+    internal static class TypeExtensions
+    {
+        public static TypeCode GetTypeCode(this Type type)
+        {
+            if(!type.IsValueType)
+            {
+                return TypeCode.Object;
+            }
+            return GetTypeCode(Activator.CreateInstance(type));
+        }
+
+        public static TypeCode GetTypeCode(object request)

```

Comment:
Why is there a non-extension method in a class that appears to be named for the extensions?